redis/src/cluster_async/mod.rs

```diff
@@ -496,47 +513,26 @@ where
         Ok(slot_map)
     }
 
-    fn get_connection(&mut self, route: &Route) -> (String, ConnectionFuture<C>) {
-        if let Some(addr) = self.slots.slot_addr_for_route(route) {
-            let addr = addr.to_string();
-            if let Some(conn) = self.connections.get(&addr) {
-                return (addr, conn.clone());
-            }
-
-            // Create new connection.
-            //
-            let (_, random_conn) = get_random_connection(&self.connections, None); // TODO Only do this lookup if the first check fails
-            let connection_future = {
-                let addr = addr.clone();
-                let params = self.cluster_params.clone();
-                async move {
-                    match connect_and_check(&addr, params).await {
-                        Ok(conn) => conn,
-                        Err(_) => random_conn.await,
-                    }
-                }
-            }
-            .boxed()
-            .shared();
-            self.connections
-                .insert(addr.clone(), connection_future.clone());
-            (addr, connection_future)
-        } else {
-            // Return a random connection
-            get_random_connection(&self.connections, None)
-        }
-    }
-
     fn try_request(
         &mut self,
         info: &RequestInfo<C>,
     ) -> impl Future<Output = (String, RedisResult<Response>)> {
         // TODO remove clone by changing the ConnectionLike trait
         let cmd = info.cmd.clone();
-        let (addr, conn) = if !info.excludes.is_empty() || info.route.is_none() {
-            get_random_connection(&self.connections, Some(&info.excludes))
+        let (addr, conn) = if info.route.is_none() {
+            get_random_connection(&self.connections)
         } else {
-            self.get_connection(info.route.as_ref().unwrap())
+            let route = info.route.as_ref().unwrap();
+            if let Some(addr) = self.slots.slot_addr_for_route(route) {
+                let addr = addr.to_string();
+                if let Some(conn) = self.connections.get(&addr) {
+                    (addr, conn.clone())
+                } else {
+                    get_random_connection(&self.connections)
+                }
+            } else {
+                get_random_connection(&self.connections)
```

Comment:
minor:
this 3-fold do {} else {get_random_connection}

makes me think that the structure of lines 522-534 can be terser. Maybe something like this:

```rs
 let (addr, conn) = info.route
   .and_then(|route| self.slots.slot_addr_for_route(route)
   .and_then(|addr| {
      let addr = addr.to_string();
      self.connections.get(&addr).map(|conn| (addr, conn))
   }).unwrap_or_else(||get_random_connection(&self.connections));
```

Reply:
Yep, looks great!